From: Michal Kubecek <mkubecek@suse.cz>
Date: Fri, 17 Sep 2021 13:20:18 +0200
Subject: nvmem: nintendo-otp: add dependency on CONFIG_HAS_IOMEM
Patch-mainline: Submitted - 2021-09-17 - 694766b15cfa3759e3686145417952d320029390.1631877302.git.mkubecek@suse.cz
References: 202108250657.h5CWR7Xf-lkp@intel.com

Function nintendo_otp_probe() calls devm_ioremap_resource() which is only
available if CONFIG_HAS_IOMEM is enabled. Add a kconfig dependency like
other similar drivers have.

Fixes: 3683b761fe3a ("nvmem: nintendo-otp: Add new driver for the Wii and Wii U OTP")
Signed-off-by: Michal Kubecek <mkubecek@suse.cz>
---
 drivers/nvmem/Kconfig | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/nvmem/Kconfig
+++ b/drivers/nvmem/Kconfig
@@ -108,6 +108,7 @@ config MTK_EFUSE
 	  will be called efuse-mtk.
 
 config NVMEM_NINTENDO_OTP
+	depends on HAS_IOMEM
 	tristate "Nintendo Wii and Wii U OTP Support"
 	help
 	  This is a driver exposing the OTP of a Nintendo Wii or Wii U console.
